tests/real/enterprise.rs

```diff
@@ -0,0 +1,321 @@
+    } else {
+        panic!("Expected protocol error.")
+    }
+
+    let had_job = consumer.run_one(0, &[queue_name]).unwrap();
+    assert!(had_job);
+    let had_another_one = consumer.run_one(0, &[queue_name]).unwrap();
+
+    // For the non-enterprise edition of Faktory, this assertion WILL FAIL:
+    assert!(!had_another_one);
+
+    // Now let's repeat the latter case, but providing different args to job2:
+    let job1 = JobBuilder::new(job_type)
+        .args(args.clone())
+        .queue(queue_name)
+        .unique_for(unique_for_secs)
+        .build();
+    producer.enqueue(job1).unwrap();
+    // this one is *NOT* a 'duplicate' ...
+    let job2 = JobBuilder::new(job_type)
+        .args(vec![Value::from("ISBN-13:9781718501850"), Value::from(101)])
+        .queue(queue_name)
+        .unique_for(unique_for_secs)
+        .build();
+    // ... so the server will accept it:
+    producer.enqueue(job2).unwrap();
+
+    let had_job = consumer.run_one(0, &[queue_name]).unwrap();
+    assert!(had_job);
+    let had_another_one = consumer.run_one(0, &[queue_name]).unwrap();
+    assert!(had_another_one);
+
+    // and the queue is empty again:
+    let had_job = consumer.run_one(0, &[queue_name]).unwrap();
+    assert!(!had_job);
+}
+
+#[test]
+fn ent_unique_job_until_success() {
+    use faktory::error;
+    use std::thread;
+    use std::time;
+
+    skip_if_not_enterprise!();
+
+    let url = learn_faktory_url();
+
+    let queue_name = "ent_unique_job_until_success";
+    let job_type = "order";
+
+    // the job will be being executed for at least 3 seconds,
+    // but is unique for 4 seconds;
+    let difficulty_level = 3;
+    let unique_for = 4;
+
+    let url1 = url.clone();
+    let handle = thread::spawn(move || {
+        // prepare producer and consumer, where the former can
+        // send a job difficulty level as a job's args and the lattter
+        // will sleep for a corresponding period of time, pretending
+        // to work hard:
+        let mut producer_a = Producer::connect(Some(&url1)).unwrap();
+        let mut consumer_a = ConsumerBuilder::default();
+        consumer_a.register(job_type, |job| -> io::Result<_> {
+            let args = job.args().to_owned();
+            let mut args = args.iter();
+            let diffuculty_level = args
+                .next()
+                .expect("job difficulty level is there")
+                .to_owned();
+            let sleep_secs =
+                serde_json::from_value::<i64>(diffuculty_level).expect("a valid number");
+            thread::sleep(time::Duration::from_secs(sleep_secs as u64));
+            Ok(eprintln!("{:?}", job))
+        });
+        let mut consumer_a = consumer_a.connect(Some(&url1)).unwrap();
+        let job = JobBuilder::new(job_type)
+            .args(vec![difficulty_level])
+            .queue(queue_name)
+            .unique_for(unique_for)
+            .unique_until_success() // Faktory's default
```

Comment:
What happens when these are both set? Does it change the behavior in a particular way? Would be worth adding to the docs (and same with `unique_until_start`).

Reply:
Added more details to the docs for both `unique_until_start` and `unique_until_success`. Besides, I added a test for unique lock bypassing, which is something I missed on the first iteration. And since it has found its place in the Faktory [docs](https://github.com/contribsys/faktory/wiki/Ent-Unique-Jobs#bypassing-uniqueness), I added a note on this to the `unique_for` setter docs.

Reply:
Perfect, thank you!